src/components/Developers/Developers.tsx

```diff
@@ -1,60 +1,85 @@
-  const { packageMeta } = React.useContext(DetailContext);
-
-  const handleLoadMore = (): void => {
-    setVisibleDevs(visibleDevs + VISIBLE_MAX);
-  };
-
-  const renderDeveloperDetails = ({ name, avatar, email }, packageMeta): JSX.Element => {
-    const { name: packageName, version } = packageMeta.latest;
-
-    return <AvatarTooltip avatar={avatar} email={email} name={name} packageName={packageName} version={version} />;
-  };
-
-  const renderDevelopers = (developers, packageMeta): JSX.Element => {
-    const listVisibleDevelopers = developers.slice(0, visibleDevs);
-
-    return (
-      <Fragment>
-        <StyledText variant={'subtitle1'}>{type}</StyledText>
-        <Content>
-          {listVisibleDevelopers.map(developer => (
-            <Details key={developer.email}>{renderDeveloperDetails(developer, packageMeta)}</Details>
-          ))}
-          {visibleDevs < developers.length && (
-            <Fab onClick={handleLoadMore} size="small">
-              <Add />
-            </Fab>
-          )}
-        </Content>
-      </Fragment>
-    );
-  };
-
-  const developerList = packageMeta && packageMeta.latest[type];
-  if (!developerList || developerList.length === 0) {
-    return null;
+export const StyledText = styled(Text)<{ theme?: Theme }>(({ theme }) => ({
+  fontWeight: theme && theme.fontWeight.bold,
+  marginBottom: '10px',
+  textTransform: 'capitalize',
+}));
+
+const StyledBox = styled(Box)({
+  '> *': {
+    margin: 5,
+  },
+});
+
+export const VISIBLE_MAX = 2;
+
+/* eslint-disable react/jsx-no-bind */
+const Developers: React.FC<Props> = ({ type, visibleMax = VISIBLE_MAX }) => {
+  const detailContext = useContext(DetailContext);
+
+  if (!detailContext) {
+    throw Error("The app's detail Context was not correct used");
   }
 
-  return renderDevelopers(developerList, packageMeta);
+  const developers = useMemo(() => getUniqueDeveloperValues(detailContext.packageMeta?.latest[type]), [
+    detailContext.packageMeta,
+    type,
+  ]);
+  const [visibleDevelopersMax, setVisibleDevelopersMax] = useState(visibleMax);
+  const [visibleDevelopers, setVisibleDevelopers] = useState(developers);
+
+  useEffect(() => {
+    if (!developers) return;
+    setVisibleDevelopers(developers.slice(0, visibleDevelopersMax));
+  }, [developers, visibleDevelopersMax]);
+
+  if (!visibleDevelopers || !developers) return null;
+
+  return (
+    <>
+      <StyledText variant={'subtitle1'}>{type}</StyledText>
+      <StyledBox display="flex" flexWrap="wrap" margin="10px 0 10px 0">
+        {visibleDevelopers.map(visibleDeveloper => (
+          <Tooltip key={visibleDeveloper.email} title={visibleDeveloper.name}>
+            <Avatar alt={visibleDeveloper.name} src={visibleDeveloper.avatar} />
+          </Tooltip>
+        ))}
+        {visibleDevelopersMax < developers.length && (
+          <Fab onClick={() => setVisibleDevelopersMax(visibleDevelopersMax + VISIBLE_MAX)} size="small">
```

Comment:
You can extract this to a function above and wrap it with `useCallback` and avoid create a closure here `() => setVisibleDevelopersMax(visibleDevelopersMax + VISIBLE_MAX)`